addons/account/models/chart_template.py

```diff
@@ -636,6 +636,21 @@ def _post_load_data(self, template_code, company, template_data):
             company.account_purchase_tax_id = self.env['account.tax'].search([
                 *self.env['account.tax']._check_company_domain(company),
                 ('type_tax_use', 'in', ('purchase', 'all'))], limit=1).id
+        # Set default taxes on products (only on products having already a tax set in another company, as some flows require no tax at all (e.g TIPS in PoS))
+        all_companies = self.env['res.company'].sudo().search([]).ids
+        # need to get all companies ids and browse the product in sudo + allowed_company_ids to check for the taxes_id and supplier_taxes_id fields
+        # that would just look empty all the time for the current user/company
+        sudoed_products = self.env['product.product'].sudo().with_context(allowed_company_ids=all_companies).search(self.env['product.product']._check_company_domain(company))
+
+        if company.account_sale_tax_id:
+            sudoed_products_sale = sudoed_products.filtered(
+                lambda p: p.taxes_id and not p.taxes_id.filtered_domain(p.taxes_id._check_company_domain(company)))
+            sudoed_products_sale.taxes_id = [Command.link(self.env.company.account_sale_tax_id.id)]
```

Comment:
If you want to use the recordset operations instead of the `write` syntax, use recordset operations :wink: 
```suggestion
            sudoed_products_sale.taxes_id += self.env.company.account_sale_tax_id
```

Reply:
:facepalm: why do I always apply comments-I-find-useless without checking on them? good catch!